src/DotLiquid/StandardFilters.cs

```diff
@@ -1048,22 +1017,47 @@ public static IEnumerable Where(IEnumerable input, string propertyName, object t
         /// <param name="targetValue">target property value</param>
         private static bool HasMatchingProperty(this object any, string propertyName, object targetValue)
         {
-            // Check if the 'any' object has a propertyName
-            object propertyValue = null;
-            if (any is IDictionary dictionary && dictionary.Contains(key: propertyName))
-            {
-                propertyValue = dictionary[propertyName];
-            }
-            else if (any != null && any.RespondTo(propertyName))
-            {
-                propertyValue = any.Send(propertyName);
-            }
-
+            var propertyValue = ResolveObjectPropertyValue(any, propertyName);
             return targetValue == null || propertyValue == null
                 ? propertyValue.IsTruthy()
                 : propertyValue.SafeTypeInsensitiveEqual(targetValue);
         }
 
+        private static object ResolveObjectPropertyValue(this object obj, string propertyName)
+        {
+            if (obj == null)
+                return null;
+            if (obj is IDictionary dictionary && dictionary.Contains(key: propertyName))
+                return dictionary[propertyName];
+            if (obj is IDictionary<string, object> dictionaryObject && dictionaryObject.ContainsKey(propertyName))
+                return dictionaryObject[propertyName];
+            var indexable = obj as IIndexable;
+            if (indexable == null)

```

Comment:
may be simplified as
```csharp
if (!(obj is IIndexable indexable))
```

This will both test and assign the new variable.

Reply:
We need variable `indexable` declared whether it is null or not. So doesn't save any lines of code.